node/src/components/consensus/protocols/highway.rs

```diff
@@ -162,7 +162,10 @@ impl<I: NodeIdT, C: Context + 'static> HighwayProtocol<I, C> {
                     past_values,
                 }]
             }
-            AvEffect::WeAreFaulty(fault) => panic!("this validator is faulty: {:?}", fault),
+            AvEffect::WeAreFaulty(fault) => {
+                error!("this validator is faulty: {:?}", fault);
+                vec![]
```

Comment:
We could also propagate it further up and let the higher level logic decide what to do about it. This will let us keep the test (relatively) as is and still have an option to panic in the higher-level logic if we choose to do so.

What do you guys think?

Reply:
I've added something here: https://github.com/CasperLabs/casper-node/pull/615/commits/a5c3555331f59e2a287344f47203666352ab13a4

As far as I can tell, we still need to stub handling our own equivocation in the era-supervisor in order for the test I've introduced to pass.

Reply:
I'm not sure I understand you Matt. I can't see anything in your new test that relies on what effects get produced.

Reply:
I wasn't sure how to test the effects, but I figured that it would be better to pass a message to the era supervisor.  What do you have in mind?

Reply:
That's what I meant – passing the effect on to the era supervisor.